stactools_sentinel2/stactools/sentinel2/stac.py

```diff
@@ -184,17 +182,22 @@ def set_asset_properties(asset):
 
     # Handle band image
 
-    band_id_search = re.search(r'_(B\w{2})_', asset_href)
+    band_id_search = re.search(r'_(B\w{2})', asset_href)
     if band_id_search is not None:
-        band_id = band_id_search.group(1)
+        band_id, href_res = os.path.splitext(asset_href)[0].split('_')[-2:]
+        asset_res = int(href_res.replace('m', ''))
         band = SENTINEL_BANDS[band_id]
+        if asset_res == BANDS_TO_RESOLUTIONS[band_id][0]:
+            asset_key = band_id
+        else:
+            asset_key = f'{band_id}_{asset_res}m'
         asset = pystac.Asset(href=asset_href,
                              media_type=asset_media_type,
                              title=band.description,
```

Comment:
Thanks @lossyrob for your work. Does it make sense to include the spatial resolution in the asset title?

Reply:
Done.

Also set the gsd correctly - gsd is based on the sensor, not spatial resolution, so that is set the same across all band assets.

Reply:
> gsd is based on the sensor, not spatial resolution, so that is set the same across all band assets.

Can you clarify this? There are no examples in the PR so it's hard to tell what the implication is without trying to follow all the code. Do you mean that the asset named `B03_60m` would have a `res` of `10` instead of `60`?

Reply:
I've got the same question @kylebarron  but wasn't quite grokking what was going on here. For the purposes of STAC, gsd is the same as what might be referred to as resolution

Reply:
Ah wait, now I see, you are referring to the actual downsampled assets.  B03 is a 10m band, but they provide 20m and 60m versions.
I personally don't see the reason for storing these downsampled files, especially with the STAC tools being developed.
But, even in the case of downsampled imagery I'd say the gsd should also change to indicate the pixel size in the asset, since for all purposes that's the effective resolution of that asset. It also clearly distinguishes it from the regular asset object.

Reply:
@matthewhanson I thought the same thing about the gsd, but https://github.com/radiantearth/stac-spec/blob/master/item-spec/common-metadata.md#gsd states that `the value of this field should be related to the spatial resolution at the sensor, rather than the pixel size of images after orthorectification, pansharpening, or scaling.`

Reply:
maybe it's time to add `resampling_factor` to the common specs ;-) 

As @kylebarron mentioned, having a GSD of 10 for band that are 20m resolution fully respect the spec but is not really user friendly. I would tend to argue that since there is already the proj extension, maybe just don't add the GSD  field which might be confusing for users.